docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1240 @@
+    },
+    "max_score": 9.712725,
+    "hits": [
+      {
+        "_index": "shakespeare",
+        "_type": "_doc",
+        "_id": "22006",
+        "_score": 9.712725,
+        "_source": {
+          "type": "line",
+          "line_id": 22007,
+          "play_name": "Antony and Cleopatra",
+          "speech_number": 12,
+          "line_number": "5.2.44",
+          "speaker": "CLEOPATRA",
+          "text_entry": "Quick, quick, good hands."
+        }
+      },
+      {
+        "_index": "shakespeare",
+        "_type": "_doc",
+        "_id": "54665",
+        "_score": 9.712725,
+        "_source": {
+          "type": "line",
+          "line_id": 54666,
+          "play_name": "Loves Labours Lost",
+          "speech_number": 21,
+          "line_number": "5.1.52",
+          "speaker": "HOLOFERNES",
+          "text_entry": "Quis, quis, thou consonant?"
+        }
+      },
+      {
+        "_index": "shakespeare",
+        "_type": "_doc",
+        "_id": "65579",
+        "_score": 9.478242,
+        "_source": {
+          "type": "line",
+          "line_id": 65580,
+          "play_name": "Merry Wives of Windsor",
+          "speech_number": 2,
+          "line_number": "3.3.2",
+          "speaker": "MISTRESS PAGE",
+          "text_entry": "Quickly, quickly! is the buck-basket--"
+        }
+      }
+    ]
+  }
+}
+```
+
+Alternatively, specify the `search_analyzer` in the mapping itself:
+
+```json
+"mappings": {
+  "properties": {
+    "text_entry": {
+      "type": "text",
+      "analyzer": "autocomplete",
+      "search_analyzer": "standard"
+    }
+  }
+}
+```
+
+### Completion suggestor (index time)
+
+Use the completion suggestor to make your autocomplete solution as efficient as possible and also to have explicit control over it’s suggestions. The completion suggestor accepts a list of suggestions and builds them into a finite-state transducer (FST), an optimized data structure that’s essentially a graph. This data structure lives in memory and is optimized for fast prefix lookups. To learn more about FSTs, see [Wikipedia](https://en.wikipedia.org/wiki/Finite-state_transducer).
+
+As the user types, Elasticsearch moves through the FST graph one character at a time along a matching path. After it runs out of user input, it examines the remaining possible endings to produce a list of suggestions.
+
+Use a dedicated field type called `completion`, which stores the FST-like data structures in the index.
+
+```json
+PUT shakespeare
+{
+  "mappings": {
+    "properties": {
+      "text_entry": {
+        "type": "completion"
+      }
+    }
+  }
+}
+```
+
+To get back suggestions, use the `search` endpoint with the `suggest` parameter:
+
+```json
+GET shakespeare/_search
+{
+  "suggest": {
+    "autocomplete": {
+      "prefix": "To be",
+      "completion": {
+        "field": "text_entry"
+      }
+    }
+  }
+}
+```
+
+The results are based on prefix matching of the FSTs.
+
+#### Sample Response
```

Comment:
Shorter response here?

Reply:
Changed all responses to show only two hits.